src/lib/db/suggest-change-store.ts

```diff
@@ -0,0 +1,123 @@
+import { ISuggestChangeStore } from '../types/stores/suggest-change-store';
+import { Logger, LogProvider } from '../logger';
+import EventEmitter from 'events';
+import { Knex } from 'knex';
+import { PartialSome } from '../types/partial';
+import { ISuggestChange, ISuggestChangeSet } from '../types/model';
+import User from '../types/user';
+import NotFoundError from '../error/notfound-error';
+
+const Tables = {
+    suggestChangeSet: 'suggest_change_set',
+};
+
+const COLUMNS = [
+    'id',
+    'environment',
+    'state',
+    'project',
+    'changes',
+    'created_by',
+    'created_at',
+    'updated_by',
+];
+
+interface ISuggestChangeSetRow {
+    id: number;
+    environment: string;
+    state?: string;
+    project?: string;
+    created_by?: string;
+    created_at?: Date;
+    changes?: ISuggestChange[];
+}
+
+export default class SuggestChangeStore implements ISuggestChangeStore {
+    private logger: Logger;
+
+    private eventBus: EventEmitter;
+
+    private db: Knex;
+
+    constructor(db: Knex, eventBus: EventEmitter, getLogger: LogProvider) {
+        this.db = db;
+        this.eventBus = eventBus;
+        this.logger = getLogger('lib/db/suggest-change-store.ts');
+    }
+
+    async create(
+        suggestChangeSet: PartialSome<ISuggestChangeSet, 'id'>,
```

Comment:
I had to go look this type up, so I might not have it completely correct. Does this make the `id` property optional on the incoming changeset? In that case, what happens to the `id` further down? If it's not there, does the DB set one automatically?

Reply:
yes the db should add it

Reply:
Thanks for the clarification!